libraries/AC_PID/AC_PID.cpp

```diff
@@ -137,15 +210,42 @@ float AC_PID::update_all(float target, float measurement, float dt, bool limit,
         _target = target;
         _error = _target - measurement;
         _derivative = 0.0f;
+        _target_derivative = 0.0f;
+#if AC_PID_ADVANCED_ENABLED
+        if (_adv_enable) {
+            if (!is_zero(_notch_T_center_freq_hz.get())) {
```

Comment:
Would it be cleaner to move this down in the notch is self?

Reply:
The parameters is in a different class - difficult to see how this is going to make things simpler